texar/modules/berts/bert_utils.py

```diff
@@ -0,0 +1,299 @@
+# Copyright 2019 The Texar Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Utils of Bert Modules.
+"""
+
+from typing import Dict, Optional
+
+import json
+import os
+import sys
+
+import torch
+import torch.nn as nn
+
+from texar.data.data_utils import maybe_download
+
+
+__all__ = [
+    "init_bert_checkpoint",
+    "load_pretrained_model",
+    "transform_bert_to_texar_config",
+]
+
+
+_BERT_PATH = "https://storage.googleapis.com/bert_models/"
+_MODEL2URL = {
+    'bert-base-uncased':
+        _BERT_PATH + "2018_10_18/uncased_L-12_H-768_A-12.zip",
+    'bert-large-uncased':
+        _BERT_PATH + "2018_10_18/uncased_L-24_H-1024_A-16.zip",
+    'bert-base-cased':
+        _BERT_PATH + "2018_10_18/cased_L-12_H-768_A-12.zip",
+    'bert-large-cased':
+        _BERT_PATH + "2018_10_18/cased_L-24_H-1024_A-16.zip",
+    'bert-base-multilingual-uncased':
+        _BERT_PATH + "2018_11_23/multi_cased_L-12_H-768_A-12.zip",
+    'bert-base-multilingual-cased':
+        _BERT_PATH + "2018_11_03/multilingual_L-12_H-768_A-12.zip",
+    'bert-base-chinese':
+        _BERT_PATH + "2018_11_03/chinese_L-12_H-768_A-12.zip",
+}
+
+
+def init_bert_checkpoint(model: nn.Module, cache_dir: str):
```

Comment:
Can users choose to disable `init_bert_checkpoint` call in the module and instead load the checkpoint by themselves?

Reply:
When the users create `model = BertClassifier(...)`, they could set `pretrained_model_name=None` and `pretrained_model_name` in `hparams` to be `None` so that `BertClassifier` won't load any pre-trained model. The users then could use `torch.load(checkpoint_file)` and `model.load_state_dict(...)` to load their own checkpoint file.